src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -788,21 +791,24 @@ export function getUsages(
         const affectTypes: string[] = [];
         if (typeName !== "") {
             affectTypes.push(typeName);
-            if (
-                effectiveType.kind === "Model" &&
-                effectiveType.templateMapper?.args
-            ) {
-                for (const arg of effectiveType.templateMapper.args) {
-                    if (
-                        arg.kind === "Model" &&
-                        "name" in arg &&
-                        arg.name !== ""
-                    ) {
-                        affectTypes.push(
-                            getFriendlyName(program, arg) ?? arg.name
-                        );
+            if (effectiveType.kind === "Model") {
+                if (effectiveType.templateMapper?.args) {
+                    for (const arg of effectiveType.templateMapper.args) {
+                        if (
+                            arg.kind === "Model" &&
+                            "name" in arg &&
+                            arg.name !== ""
+                        ) {
+                            affectTypes.push(
+                                getFriendlyName(program, arg) ?? arg.name
```

Comment:
is there a method in TCGC that could replace the getFriendlyName method here from the compiler?
And I asusme the TCGC common interface refactor could replace all of these, therefore I think it is OK if we just use this as a temporary solution and defer to the common interface refactor

Reply:
Yes. TCGC will handle all the Naming and Usage of a model. Those logic will remove after we adopt TCGC common interface.